### PR TITLE
Issue #2657558 by Devin Carlson: Required attributes wildcard in CKEditor plugin not compatible with filter system.

### PR DESCRIPTION
**Pull request by DevinCarlson**

### CODE REVIEW

File: js/plugins/drupalurl/plugin.js [hunk 1]
```diff
@@ -27,8 +27,8 @@
 
       // Generic command for adding/editing entities of all types.
       editor.addCommand('editdrupalurl', {
-        allowedContent: 'drupal-url[*]',
-        requiredContent: 'drupal-url[*]',
+        allowedContent: 'drupal-url[data-embed-url,data-url-provider]',
+        requiredContent: 'drupal-url[data-embed-url,data-url-provider]',
```
Comment (davereid) [inline js/plugins/drupalurl/plugin.js]: data-url-provider is not a required attribute to render content, it's just additional data.

File: js/plugins/drupalurl/plugin.js [hunk 2]
```diff
@@ -83,8 +83,8 @@
       // Register the URL embed widget.
       editor.widgets.add('drupalurl', {
         // Minimum HTML which is required by this widget to work.
-        allowedContent: 'drupal-url[*]',
-        requiredContent: 'drupal-url[*]',
+        allowedContent: 'drupal-url[data-embed-url,data-url-provider]',
+        requiredContent: 'drupal-url[data-embed-url,data-url-provider]',
```
Comment (davereid) [inline js/plugins/drupalurl/plugin.js]: Same here.

File: js/plugins/drupalurl/plugin.js (hunk 1 above)
Comment (DevinCarlson) [inline js/plugins/drupalurl/plugin.js]: Removed `data-url-provider`.

**Comment by slashrsm** [comment]
We have a failing test.

**Comment by DevinCarlson** [comment]
The failing test is caused by https://www.drupal.org/node/2652142 which is a core issue being worked on in https://www.drupal.org/node/2652068.

I've updated the PR with a simple workaround which can be removed once the core issue is fixed.

**Comment by slashrsm** [comment]
Merged. Thanks!

**Pull request merged**